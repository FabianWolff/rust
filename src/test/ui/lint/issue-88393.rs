// Regression test for issue #88393.

#![allow(deprecated, invalid_value, unused_must_use)]
#![forbid(unreachable_code)]

enum Void {}

fn foo() {
    if false {
        unsafe { std::mem::uninitialized::<Void>(); }
    }
    println!();
}

fn bar() {
    let b = false;
    match b {
        false => unsafe { std::mem::uninitialized::<Void>(); }
        _ => unreachable!(),
    }
    println!();
    //~^ ERROR: unreachable expression
}

fn main() {}
