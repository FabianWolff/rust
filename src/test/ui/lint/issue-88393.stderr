error: unreachable expression
  --> $DIR/issue-88393.rs:21:5
   |
LL | /     match b {
LL | |         false => unsafe { std::mem::uninitialized::<Void>(); }
LL | |         _ => unreachable!(),
LL | |     }
   | |_____- any code following this expression is unreachable
LL |       println!();
   |       ^^^^^^^^^^ unreachable expression
   |
note: the lint level is defined here
  --> $DIR/issue-88393.rs:4:11
   |
LL | #![forbid(unreachable_code)]
   |           ^^^^^^^^^^^^^^^^
note: all arms of this `match` expression diverge
  --> $DIR/issue-88393.rs:17:5
   |
LL | /     match b {
LL | |         false => unsafe { std::mem::uninitialized::<Void>(); }
LL | |         _ => unreachable!(),
LL | |     }
   | |_____^
   = note: this error originates in the macro `$crate::format_args` (in Nightly builds, run with -Z macro-backtrace for more info)

error: aborting due to previous error

